trieste/models/interfaces.py

```diff
@@ -75,6 +85,10 @@ def sample(self, query_points: TensorType, num_samples: int) -> TensorType:
         """
         raise NotImplementedError
 
+    @check_shapes(
+        "query_points: [broadcast batch..., D]",
+        "return: [batch..., E...]",
+    )
```

Comment:
Should this be returning two tensors, for mean and variance?
```suggestion
    @check_shapes(
        "query_points: [broadcast batch..., D]",
        "return[0]: [batch..., E...]",
        "return[1]: [batch..., E...]",
    )
```

Reply:
Good catch. I'll investigate why the predict_y unit tests didn't catch this!

Reply:
Looks like the previous code was treating the tuple as a single tensor with an additional leading dimension of 2 as far as shape checking is concerned. Which kinda of makes sense: e.g. you can use this to specify that a function returns a list of certain size.

I've fixed it so that the two checks are specified separately like you suggested, though it's also possible to specify it more succinctly (but less clearly) as  `"return: [2, batch..., E...]"`.